ImagifyPartner.php

```diff
@@ -0,0 +1,877 @@
+<?php
+/**
+ * Tool allowing 3rd party WordPress plugins to handle partnership with Imagify.
+ *
+ * @package WPMedia\ImagifyPartner
+ */
+
+namespace WPMedia\ImagifyPartner;
+
+if ( class_exists( 'WPMedia\ImagifyPartner\ImagifyPartner' ) || class_exists( 'Imagify_Partner' ) ) {
+	return;
+}
```

Comment:
I wanted to make sure we don't conflict with the first version of the class that is already used elsewhere, do you think that's fine, or is there a better approach?

Reply:
Yes, but it could be simplified to just the 2nd part of the conditional:

if the first version of the class exists in memory, bail out.

Reply:
If the old class is loaded after the new one...

Reply:
that's a good point, would `class_alias()` prevent an error?

Reply:
Nice idea. I think this should work:
```php
<?php
namespace WPMedia\ImagifyPartner;

if ( class_exists( 'WPMedia\ImagifyPartner\ImagifyPartner' ) || class_exists( 'Imagify_Partner' ) ) {
	return;
}

class ImagifyPartner {
	// ...
}

class_alias( '\\WPMedia\\ImagifyPartner\\ImagifyPartner', '\\Imagify_Partner' );
```
So the class `Imagify_Partner` will exist and the old version will not be loaded later.